plugin/trino-databend/src/main/java/io/trino/plugin/databend/DatabendClient.java

```diff
@@ -0,0 +1,745 @@
+import static io.trino.spi.type.Timestamps.NANOSECONDS_PER_MICROSECOND;
+import static io.trino.spi.type.TinyintType.TINYINT;
+import static io.trino.spi.type.VarcharType.createUnboundedVarcharType;
+import static io.trino.spi.type.VarcharType.createVarcharType;
+import static java.lang.Float.floatToRawIntBits;
+import static java.lang.Math.floorDiv;
+import static java.lang.Math.floorMod;
+import static java.lang.Math.max;
+import static java.lang.Math.toIntExact;
+import static java.lang.String.format;
+import static java.lang.String.join;
+import static java.time.ZoneOffset.UTC;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.joining;
+
+public class DatabendClient
+        extends BaseJdbcClient
+{
+    private static final DecimalType UINT64_TYPE = createDecimalType(20, 0);
+
+    // An empty character means that the table doesn't have a comment in Databend
+    private static final String NO_COMMENT = "";
+    private final Type jsonType;
+
+    public static final int DEFAULT_DOMAIN_COMPACTION_THRESHOLD = 1_000;
+
+    private final AggregateFunctionRewriter<JdbcExpression, ?> aggregateFunctionRewriter;
+
+    @Inject
+    public DatabendClient(BaseJdbcConfig config, ConnectionFactory connectionFactory, QueryBuilder queryBuilder, TypeManager typeManager, IdentifierMapping identifierMapping, RemoteQueryModifier queryModifier)
+    {
+        super("\"", connectionFactory, queryBuilder, config.getJdbcTypesMappedToVarchar(), identifierMapping, queryModifier, false);
+        JdbcTypeHandle bigintTypeHandle = new JdbcTypeHandle(Types.BIGINT, Optional.of("bigint"), Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty());
+        ConnectorExpressionRewriter<ParameterizedExpression> connectorExpressionRewriter = JdbcConnectorExpressionRewriterBuilder.newBuilder().addStandardRules(this::quoted).build();
+        this.jsonType = typeManager.getType(new TypeSignature(StandardTypes.JSON));
+        this.aggregateFunctionRewriter = new AggregateFunctionRewriter<>(connectorExpressionRewriter, ImmutableSet.<AggregateFunctionRule<JdbcExpression, ParameterizedExpression>>builder()
+                .add(new ImplementCountAll(bigintTypeHandle))
+                .add(new ImplementCount(bigintTypeHandle))
+                .add(new ImplementCountDistinct(bigintTypeHandle, false))
+                .add(new ImplementMinMax(false))
+                .add(new ImplementSum(DatabendClient::toTypeHandle))
+                .add(new ImplementAvgFloatingPoint())
+                .add(new ImplementAvgBigint())
+                .add(new ImplementCorr())
+                .add(new ImplementCovarianceSamp())
+                .add(new ImplementStddevSamp())
+                .add(new ImplementStddevPop())
+                .add(new ImplementVarianceSamp())
+                .add(new ImplementVariancePop())
+                .add(new ImplementCovariancePop()).build());
```

Comment:
Are all these aggregation pushdown ensured by tests?

Reply:
I remove these pushdown and support it in another pr after this merged. Thank u.